src/Classes/ExtBuildListControl.lua

```diff
@@ -0,0 +1,551 @@
+				self:DrawString(x + authorWidth + 20, currentHeight, "LEFT", 14, self.font, s_format('%s', build.version))
+				SetDrawColor(0.5, 0.5, 0.5)
+				if authorWidth then
+					self:DrawImage(nil, x + authorWidth + 10, currentHeight - 7, 1, 28)
+					self:DrawImage(nil, x + authorWidth + DrawStringWidth(14, self.font, build.version) + 30, currentHeight - 7, 1, 28)
+				end
+			end
+
+			currentHeight = currentHeight + 20
+
+			-- decorator line
+			SetDrawColor(0.5, 0.5, 0.5)
+			self:DrawImage(nil, x - 9, currentHeight, self.width(), 1)
+			currentHeight = currentHeight + 8
+			SetDrawColor(1, 1, 1)
+
+			-- stats
+			local dpsText = "DPS: 0"
+			local lifeText = "Life: 0"
+			local ehpText = "EHP: 0"
+			if build.dps then
+				-- SetDrawColor(1, 0, 0)
+				dpsText = formatNumSep(s_format('DPS: %0.f', build.dps))
+			end
+			if build.life or build.es then
+				-- SetDrawColor(0, 1, 0)
+				lifeText = formatNumSep(s_format('%s: %0.f', build.life > build.es and "Life" or "ES", math.max(build.life, build.es)))
+			end
+			if build.ehp then
+				-- SetDrawColor(0, 0, 1)
+				ehpText = formatNumSep(s_format('EHP: %0.f', build.ehp))
+			end
+
+			-- prevent overlapping on smaller screens.
+			local dpsWidth = DrawStringWidth(14, self.font, dpsText)
+			local lifeWidth = DrawStringWidth(14, self.font, lifeText)
+			local ehpWidth = DrawStringWidth(14, self.font, ehpText)
+			if (dpsWidth + lifeWidth + ehpWidth < self.width() - 30) then
+				self:DrawString(x, currentHeight, "LEFT", 14, self.font, dpsText)
+				SetDrawColor(0.5, 0.5, 0.5)
+				self:DrawImage(nil, x + dpsWidth + 10, currentHeight - 7, 1, 28)
+				SetDrawColor(1, 1, 1)
+				self:DrawString(x + dpsWidth + 20, currentHeight, "LEFT", 14, self.font, lifeText)
+				SetDrawColor(0.5, 0.5, 0.5)
+				self:DrawImage(nil, x + dpsWidth + lifeWidth + 30, currentHeight - 7, 1, 28)
+				self:DrawImage(nil, x + dpsWidth + lifeWidth + ehpWidth + 50, currentHeight - 7, 1, 28)
+				SetDrawColor(1, 1, 1)
+				self:DrawString(x + dpsWidth + lifeWidth + 40, currentHeight, "LEFT", 14, self.font, ehpText)
+				currentHeight = currentHeight + 20
+				-- decorator line
+				SetDrawColor(0.5, 0.5, 0.5)
+				self:DrawImage(nil, x - 9, currentHeight, self.width(), 1)
+				currentHeight = currentHeight + 3
+			end
+
+			-- SetDrawColor(1, 1, 1)
+
+			if build.metadata then
+				currentHeight = currentHeight + 4
+				for _, metadata in pairs(build.metadata) do
+					SetDrawColor(1, 1, 1)
+					self:DrawString(x, currentHeight, "LEFT", 14, self.font, metadata.key .. ": " .. metadata.value)
+					currentHeight = currentHeight + 20
+					SetDrawColor(0.5, 0.5, 0.5)
+					self:DrawImage(nil, x - 9, currentHeight, self.width(), 1)
+				end
+
+			end
+
+			-- import button
+			local importButton = {
+				buildLink = build.buildLink,
+				buildName = build.buildName,
+				authorName = build.author,
+				x0 = x,
+				y0 = currentHeight + 6,
+				x1 = x + 47,
+				y1 = currentHeight + 26
+			}
```

Comment:
Why not just create actual buttons here?  They come with everything you need, like placement, width, height, click handler, etc.  I'm not immediately seeing anything that requires a bespoke solution like this.

Reply:
I thought about implementing a build card and drawing each build as its own control class, that way buttons would be also taken care of with buttoncontrol. But integrating that with scrollbar class would cause issues such as partial drawing so I left as is. Never thought about only converting buttons though, I will try it now.

Reply:
Spent whole day trying to create a better way (card class), then a few hours on implementing just the buttons but its finally fixed! Main reason that made it a hassle was anchoring/preventing overflows while keeping it under controls table.